### ISSUE TITLE
false positive of redundant do when performing side effect

### ISSUE THREAD

**Issue by mk** [clj-kondo/clj-kondo#1898]
**version**
clj-kondo --version
clj-kondo v2022.11.02

**problem & repro**
I have the following code in Clerk:
```clojure
(defn stdout-reporter [build-event]
  (doto (describe-event build-event)
    (print)
    (do (flush))))
```

which produces this warning:
```sh
$ clj-kondo --lint src/nextjournal/clerk/builder.clj
src/nextjournal/clerk/builder.clj:95:5: warning: redundant do
```
**expected behavior**

Since `flush` doesn't take any arguments, I'm using the `do` to perform this side-effect. Should I be using a different pattern or should clj-kondo not report a redundant do in this case?

**Comment by NoahTheDuke** [comment]
```clojure
(defn stdout-reporter [build-event]
  (doto (describe-event build-event)
    (print))
  (flush))
```

**Comment by mk** [comment]
@NoahTheDuke I want to return the value of `describe-event` from the call to `stdout-reporter`.

**Comment by NoahTheDuke** [comment]
```clojure
(defn stdout-reporter [build-event]
  (let [result (describe-event build-event)]
    (print result)
    (flush)
    result))
```

**Comment by borkdude** [comment]
`doto` is meant for doing things to something. `(flush)` isn't doing anything to your something, I'd say you're slightly abusing `doto` here, which imo is not something clj-kondo should account for. Just use a `let` I'd say.

**Comment by mk** [comment]
Alright, probably clearer as well.

**Comment by mk** [comment]
Didn't like to introduce a new name so I ended up going with another name:
```clojure
(defn ^:private print+flush [x]
  (print x)
  (flush))

(defn stdout-reporter [build-event]
  (doto (describe-event build-event)
    print+flush))
```
